src/components/Layout/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import Image from 'next/image';
+
+import Avatar from './Avatar';
+import Sidebar from './Sidebar';
+import { UserProfile } from 'services/user';
+
+interface ComponentProps {
+  user?: UserProfile | null;
+  children: React.ReactNode;
+}
+
+const Layout = ({ user, children }: ComponentProps) => {
+  const [showSidebar, setShowSidebar] = useState(false);
+
```

Comment:
Should we add a guard here to render empty when `user === null`?
Then we don't need to check the `user` when rendering

Reply:
We can't render empty here because this component still renders the Nimble logo and the children even though the user is null.

The only parts that is not render when user is null are avatar and sidebar